src/drivers/mb12xx/mb12xx.cpp

```diff
@@ -576,47 +661,60 @@ MB12XX::stop()
 void
 MB12XX::cycle_trampoline(void *arg)
 {
+
 	MB12XX *dev = (MB12XX *)arg;
 
 	dev->cycle();
+
 }
 
 void
 MB12XX::cycle()
 {
-	/* collection phase? */
 	if (_collect_phase) {
+		_index_counter = addr_ind[_cycle_counter]; /*sonar from previous iteration collect is now read out */
+		set_address(_index_counter);
 
 		/* perform collection */
 		if (OK != collect()) {
 			log("collection error");
-			/* restart the measurement state machine */
+			/* if error restart the measurement state machine */
 			start();
 			return;
 		}
 
 		/* next phase is measurement */
 		_collect_phase = false;
-
-		/*
-		 * Is there a collect->measure gap?
-		 */
-		if (_measure_ticks > USEC2TICK(MB12XX_CONVERSION_INTERVAL)) {
+		
+		/* change i2c adress to next sonar */
+		_cycle_counter=_cycle_counter+1;
+		if (_cycle_counter >= addr_ind.size()){
+		_cycle_counter=0;
+		}
+		
+		/* Is there a collect->measure gap? Yes, and the timing is set equal to the cycling_rate
+		   Otherwise the next sonar would fire without the first one having received its reflected sonar pulse */
+		
+		if (_measure_ticks > USEC2TICK(_cycling_rate)) {
 
 			/* schedule a fresh cycle call when we are ready to measure again */
 			work_queue(HPWORK,
 				   &_work,
 				   (worker_t)&MB12XX::cycle_trampoline,
 				   this,
-				   _measure_ticks - USEC2TICK(MB12XX_CONVERSION_INTERVAL));
-
+				   _measure_ticks - USEC2TICK(_cycling_rate));
 			return;
 		}
 	}
+	/* Measurement (firing) phase */
 
-	/* measurement phase */
+	/* ensure sonar i2c adress is still correct */
+	_index_counter = addr_ind[_cycle_counter]; 
+	set_address(_index_counter);
+	
+	/* Perform measurement */
 	if (OK != measure()) {
-		log("measure error");
+		log("measure error sonar adress %d",_index_counter);
```

Comment:
Please change to debug(), since it will otherwise flood the console if a cable breaks. Alternatively do an error counter and only report once every N counts or if the error count exceeds a certain number, and just once.